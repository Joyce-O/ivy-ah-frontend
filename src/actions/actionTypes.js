export const AUTH_LOADING = 'AUTH_LOADING';
export const LOGIN = 'LOGIN';
export const TOGGLE_MODAL = 'TOGGLE_MODAL';
export const FETCH_ARTICLE_STARTED = 'FETCH_ARTICLE_STARTED';
export const FETCH_ARTICLE_SUCCEEDED = 'FETCH_ARTICLE_SUCCEEDED';
export const FETCH_ARTICLE_FAILED = 'FETCH_ARTICLE_FAILED';
export const EDIT_BOOKMARKS_STARTED = 'EDIT_BOOKMARKS_STARTED';
export const FETCH_BOOKMARKS_SUCCEEDED = 'FETCH_BOOKMARKS_SUCCEEDED';
export const ADD_BOOKMARKS_SUCCEEDED = 'ADD_BOOKMARKS_SUCCEEDED';
export const REMOVE_BOOKMARKS_SUCCEEDED = 'REMOVE_BOOKMARKS_SUCCEEDED';
export const EDIT_BOOKMARKS_FAILED = 'EDIT_BOOKMARK_FAILED';
export const AUTH_SUCCESS = 'AUTH_SUCCESS';
export const AUTH_FAIL = 'AUTH_FAIL';
export const VERIFY_EMAIL = 'VERIFY_EMAIL';
