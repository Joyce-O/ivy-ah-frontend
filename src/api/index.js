import axios from 'axios';

const API_BASE_URL = 'https://ivy-ah-backend-staging.herokuapp.com/api/v1';

const token = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6ImFuZGVsYVM2NUBkYW5kZWxhLmNvbSIsImlkIjoiMmM0YTRlMjMtN2FmYS00Y2Y4LTg0ZjMtZDVhNDFkYjAxNGMyIiwicm9sZSI6InVzZXIiLCJpYXQiOjE1NTgwMzY5OTh9.rrk2ZMbDzQvAwy9-XHPruTAkFo0rZkmEoT19W-loEBE';

export const client = axios.create({
  baseURL: API_BASE_URL,
  headers: {
    'Content-Type': 'application/json',
    Authorization: token
  },
});

const urlParams = new URLSearchParams(window.location.search);
const resetToken = urlParams.get('resetToken');
export const signup = params => client.post('/users/signup', params);
export const login = params => client.post('/users/login', params);
export const forgotPassword = params => client.post('/users/forgotpassword', params);
export const resetPassword = params => axios({
  method: 'patch',
  url: `${API_BASE_URL}/users/resetpassword?resetToken=${resetToken}`,
  data: params
});
export const fetchBookmarks = () => client.get('/bookmarks');
export const addBookmark = article => client.post('/bookmarks', { article });
export const removeBookmark = article => client.delete('/bookmarks', {
  data: {
    article
  }
});
export const fetchArticle = id => client.get(`/articles/${id}`);
export const fetchProfile = id => client.get(`/profiles/${id}`);
export const fetchUserArticle = userId => client.get(`users/articles/${userId}`);
export const fetchUserFollowers = authorId => client.get(`profiles/${authorId}/followers`);
