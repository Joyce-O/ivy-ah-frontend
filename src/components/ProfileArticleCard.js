import React from 'react';
import { Card, Image } from 'semantic-ui-react';

const ProfileArticleCard = () => (
  <div className="profile-article">
    <Card.Group>
      <div className="profile-article-wrapper">
        <Image
          className="profile-article-user-image"
          src="https://images.pexels.com/photos/555790/pexels-photo-555790.png?cs=srgb&dl=black-and-white-boy-casual-555790.jpg&fm=jpg"
          avatar
          floated="left"
        />
        <span className="profile-article-user-name">Daddy Kossy supreme</span>
        <p className="profile-article-readtime"> 2 min read</p>
      </div>
      <div
        style={{
          overflow: 'hidden',
          minWidth: '100%',
          height: '10rem',
          backgroundImage:
            "url('https://images.pexels.com/photos/2259810/pexels-photo-2259810.jpeg?cs=srgb&dl=adventure-alps-clouds-2259810.jpg&fm=jpg')",
          backgroundRepeat: 'no-repeat',
          backgroundSize: 'cover',
          backgroundPosition: 'center'
        }}
      >
        `
      </div>
      <div className="w-100">
        <div className="detail-container">
          <p>awesome article</p>
          <div className="article-hypes">70 hypes</div>
        </div>
      </div>
    </Card.Group>
  </div>
);

export default ProfileArticleCard;
