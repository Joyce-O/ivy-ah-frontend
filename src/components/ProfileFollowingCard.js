import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Card, Image, Grid } from 'semantic-ui-react';

const ProfileFollowingCard = ({ follower }) => (
  <Grid.Column>
    <Link href="profileRedirect" to={`/profileRedirect/${follower.id}`}>
      <div style={{ marginBottom: '2rem' }}>
        <Card fluid>
          <Card.Content>
            <Image
              style={{ width: '6rem', marginRight: '2rem' }}
              floated="left"
              size="small"
              circular
              src={
            follower.image
              ? follower.image
              : `https://ui-avatars.com/api/?bold=true&background=3157BE&color=fff&name=+${
                follower.lastname
              }+${follower.firstname}`
          }
            />
            <Card.Header
              style={{ paddingTop: '0.7rem', paddingBottom: '0.3rem' }}
            >
              {`${follower.lastname} ${follower.firstname}`}
            </Card.Header>
            <Card.Meta style={{ paddingBottom: '0.5rem' }}>
              {'joined Authors\' haven '}
            </Card.Meta>
            <Card.Meta>Following</Card.Meta>
          </Card.Content>
        </Card>
      </div>
    </Link>
  </Grid.Column>
);

ProfileFollowingCard.propTypes = {
  follower: PropTypes.shape([])
};

ProfileFollowingCard.defaultProps = {
  follower: []
};

export default ProfileFollowingCard;
