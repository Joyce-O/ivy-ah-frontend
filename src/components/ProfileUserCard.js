import React from 'react';
import { Image, Header, Button } from 'semantic-ui-react';

const ProfileUserCard = () => (
  <div className="profile">
    <Image
      style={{ margin: '0 auto' }}
      src="https://images.pexels.com/photos/2080383/pexels-photo-2080383.jpeg?cs=srgb&dl=adult-facial-expression-male-2080383.jpg&fm=jpg"
      size="medium"
      circular
      fluid
    />
    <Header className="profile-username" size="huge">
      Daddy Kossy Supreme
    </Header>
    <p>
      Been on Authors&apos; Haven
      <span style={{ display: 'inline-block' }}>since December 10, 2018</span>
    </p>
    <Button className="btn blueButton">Edit profile</Button>
  </div>
);

export default ProfileUserCard;
