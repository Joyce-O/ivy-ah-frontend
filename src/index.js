import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import 'semantic-ui-css/semantic.min.css';
import 'overlayscrollbars/css/OverlayScrollbars.min.css';
import 'react-draft-wysiwyg/dist/react-draft-wysiwyg.css';
import App from './containers/App';
import Store from './store';
import './assets/sass/index.scss';

render(<Provider store={Store}><App /></Provider>, document.getElementById('root'));
