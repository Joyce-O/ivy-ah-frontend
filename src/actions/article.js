import * as api from '../api';
import * as actions from './actionTypes';

export const fetchArticlesStarted = () => ({
  type: actions.FETCH_ALL_ARTICLES_STARTED,
});

const fetchArticlesSucceeded = articles => ({
  type: actions.FETCH_ALL_ARTICLES_SUCCEEDED,
  payload: { articles }
});

const fetchArticlesFailed = err => ({
  type: actions.FETCH_ALL_ARTICLES_FAILED,
  payload: { err }
});

export const fetchArticles = (page, limit) => (dispatch) => {
  dispatch(fetchArticlesStarted());
  return api.fetchArticles(page, limit).then((res) => {
    dispatch(fetchArticlesSucceeded(res.data));
  }).catch((err) => {
    dispatch(fetchArticlesFailed(err.message));
  });
};
