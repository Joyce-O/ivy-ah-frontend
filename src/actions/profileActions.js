import * as actions from './actionTypes';
import { fetchProfile, fetchUserArticle, fetchUserFollowers } from '../api';


export const getProfileStart = () => ({
  type: actions.GET_PROFILE_STARTED
});

export const getProfileSucceeded = profile => ({
  type: actions.GET_PROFILE_SUCCEEDED,
  profile
});


export const getProfileFailed = () => ({
  type: actions.GET_PROFILE_FAILED
});

export const getProfileContentStart = () => ({
  type: actions.GET_PROFILE_CONTENT_START
});

export const getProfileContentSucceeded = (content, contentType) => ({
  type: actions.GET_PROFILE_CONTENT_SUCCEEDED,
  content,
  contentType
});

export const getProfileContentFailed = () => ({
  type: actions.GET_PROFILE_CONTENT_FAILED
});

export const getUserArticle = userId => (dispatch) => {
  dispatch(getProfileContentStart());
  return fetchUserArticle(userId)
    .then((response) => {
      const userArticles = response.data;
      dispatch(getProfileContentSucceeded(userArticles, 'articles'));
    })
    .catch((err) => {
      dispatch(getProfileContentFailed());
    });
};

export const getUserFollowers = authorId => (dispatch) => {
  dispatch(getProfileContentStart());
  return fetchUserFollowers(authorId)
    .then((response) => {
      const userFollowers = response.data.data[0].followers;
      dispatch(getProfileContentSucceeded(userFollowers, 'followers'));
    })
    .catch((err) => {
      dispatch(getProfileContentFailed());
    });
};

export const getProfile = id => (dispatch) => {
  dispatch(getProfileStart());
  return fetchProfile(id)
    .then((response) => {
      const { profile } = response.data;
      dispatch(getProfileSucceeded(profile));
    })
    .catch((err) => {
      dispatch(getProfileFailed());
    });
};
