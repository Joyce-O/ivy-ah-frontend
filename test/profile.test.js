import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import { Grid } from 'semantic-ui-react';
import ProfileReducer from '../src/reducers/profile';
import * as api from '../src/api';
import ProfileArticleCard from '../src/components/ProfileArticleCard';
import ProfileFollowingCard from '../src/components/ProfileFollowingCard';
import ProfileUserCard from '../src/components/ProfileUserCard';
import * as actions from '../src/actions/profileActions';
import * as types from '../src/actions/actionTypes';

const mockStore = configureStore([thunk]);

describe('<ProfileArticleCard/>', () => {
  const article = {
    id: 'someid',
    user: {
      image: 'someUrl',
      firtname: 'someName',
      lastname: 'anothername'
    },
    readTime: '7 sec'
  };
  it('should return a <Grid.Column> </Grid.Column> component', () => {
    const wrapper = shallow(<ProfileArticleCard article={article} />);
    expect(wrapper.find(<Grid.Column />));
  });

  it('Should match the snapshot', () => {
    const tree = renderer.create(<MemoryRouter>
      <ProfileArticleCard article={article} />
    </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<ProfileFollowingCard/>', () => {
  const follower = {
    lastname: 'somelastname',
    firstname: 'anyname',
    image: 'someurl',
    id: 'someid'
  };
  it('should return a <Grid.Column> </Grid.Column> component', () => {
    const wrapper = shallow(<ProfileFollowingCard follower={follower} />);
    expect(wrapper.find(<Grid.Column />));
  });

  it('Should match the snapshot', () => {
    const tree = renderer
      .create(<MemoryRouter>
        <ProfileFollowingCard follower={follower} />
      </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<ProfileFollowingCard/>', () => {
  const profile = {
    lastname: 'somelastname',
    firstname: 'anyname',
    image: 'someurl',
    id: 'someid'
  };

  const signedInUser = 'someotherid';
  it('should return a <Grid.Column> </Grid.Column> component', () => {
    const wrapper = shallow(<ProfileUserCard profile={profile} signedInUser={signedInUser} />);
    expect(wrapper.find(<Grid.Column />));
  });

  it('Should match the snapshot', () => {
    const tree = renderer
      .create(<ProfileUserCard profile={profile} signedInUser={signedInUser} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Profile reducers', () => {
  const initialState = {
    user: {},
    profileContent: null,
    loadingProfile: false,
    loadingContent: false,
    loadedContentType: 'articles',
    loadingFailed: false,
    error: null
  };

  it('should return the initial state', () => {
    expect(ProfileReducer(undefined, {})).toEqual({
      user: {},
      profileContent: null,
      loadingProfile: false,
      loadingContent: false,
      loadedContentType: 'articles',
      loadingFailed: false,
      error: null
    });
  });

  it('should change loadingProfile to true', () => {
    expect(ProfileReducer(initialState, actions.getProfileStart())).toEqual({
      ...initialState,
      loadingProfile: true
    });
  });

  it('should change loadingContent to false', () => {
    expect(ProfileReducer(initialState, actions.getProfileContentFailed())).toEqual({
      ...initialState,
      loadingContent: false
    });
  });

  it('should change loadingContent to true', () => {
    expect(ProfileReducer(initialState, actions.getProfileContentStart())).toEqual({
      ...initialState,
      loadingContent: true
    });
  });

  it('should change load a new profile', () => {
    expect(ProfileReducer(initialState, actions.getProfileSucceeded())).toEqual({
      ...initialState,
      user: actions.profile,
      loadingProfile: false
    });
  });

  it('should change load a new content', () => {
    expect(ProfileReducer(initialState, actions.getProfileContentSucceeded())).toEqual({
      ...initialState,
      loadingContent: false,
      profileContent: actions.content,
      loadedContentType: actions.contentType
    });
  });


  it('should change loadingFailed to true and loadingProfile to false', () => {
    expect(ProfileReducer(initialState, actions.getProfileFailed())).toEqual({
      ...initialState,
      loadingProfile: false,
      loadingFailed: true
    });
  });
});


describe('Profile actions', () => {
  it('should create an action of type GET_PROFILE_STARTED', () => {
    const expectedAction = {
      type: types.GET_PROFILE_STARTED
    };
    expect(actions.getProfileStart()).toEqual(expectedAction);
  });

  it('should create an action of type GET_PROFILE_SUCCEEDED', () => {
    const expectedAction = {
      type: types.GET_PROFILE_SUCCEEDED,
      profile: 'some-profile'
    };
    expect(actions.getProfileSucceeded('some-profile')).toEqual(expectedAction);
  });

  it('should create an action of type GET_PROFILE_FAILED', () => {
    const expectedAction = {
      type: types.GET_PROFILE_FAILED,
    };
    expect(actions.getProfileFailed()).toEqual(expectedAction);
  });

  it('should create an action of type GET_PROFILE_CONTENT_FAILED', () => {
    const expectedAction = {
      type: types.GET_PROFILE_CONTENT_FAILED,
    };
    expect(actions.getProfileContentFailed()).toEqual(expectedAction);
  });

  it('should create an action of type GET_PROFILE_CONTENT_START', () => {
    const expectedAction = {
      type: types.GET_PROFILE_CONTENT_START,
    };
    expect(actions.getProfileContentStart()).toEqual(expectedAction);
  });

  it('should create an action of type GET_PROFILE_CONTENT_START', () => {
    const expectedAction = {
      type: types.GET_PROFILE_CONTENT_START,
    };
    expect(actions.getProfileContentStart()).toEqual(expectedAction);
  });

  it('should create an action of type GET_PROFILE_CONTENT_SUCCEEDED', () => {
    const expectedAction = {
      type: types.GET_PROFILE_CONTENT_SUCCEEDED,
      content: 'some-content',
      contentType: 'some-contentType',
    };
    expect(actions.getProfileContentSucceeded('some-content', 'some-contentType')).toEqual(expectedAction);
  });

  it('should create an action of type GET_PROFILE_CONTENT_FAILED', () => {
    const expectedAction = {
      type: types.GET_PROFILE_CONTENT_FAILED,
    };
    expect(actions.getProfileContentFailed()).toEqual(expectedAction);
  });

  it('should create an action of type GET_PROFILE_CONTENT_STARTED and GET_PROFILE_CONTENT_SUCCEEDED if successful', () => {
    api.client.get = jest.fn().mockReturnValue(Promise.resolve({
      data: {
        body: JSON.stringify({}),
        data: {
          articleCount: 6,
          articles: [{ }, { }, { }, {}, { }],
          currentPage: 1,
          pageCount: 1,
          status: 200
        }
      }
    }));

    const expectedActions = [
      'GET_PROFILE_CONTENT_START',
      'GET_PROFILE_CONTENT_SUCCEEDED'
    ];
    const store = mockStore({});
    return store.dispatch(actions.getUserArticle('e25af1ae-3835-46cb-a488-716936a07ab5'))
      .then(() => {
        const actualActions = store.getActions().map(action => action.type);
        expect(actualActions).toEqual(expectedActions);
      });
  });

  it('should create an action of type GET_PROFILE_CONTENT_START & GET_PROFILE_CONTENT_FAILED', () => {
    api.client.get = jest.fn().mockReturnValue(Promise.reject(new Error('An error occured')));
    const expectedActions = [
      'GET_PROFILE_CONTENT_START',
      'GET_PROFILE_CONTENT_FAILED',
    ];
    const store = mockStore({});
    return store.dispatch(actions.getUserArticle('2aec1bef'))
      .then(() => {
        const actualActions = store.getActions().map(action => action.type);
        expect(actualActions).toEqual(expectedActions);
      });
  });

  it('should create an action of type GET_PROFILE_CONTENT_STARTED and GET_PROFILE_CONTENT_SUCCEEDED if successful', () => {
    api.client.get = jest.fn().mockReturnValue(Promise.resolve({
      data: {
        data: [
          {
            authorId: 'c1f9596a-922c-41fc-95fd-513a21bd688d',
            count: 1,
            followers: []
          }
        ]
      }

    }));

    const expectedActions = [
      'GET_PROFILE_CONTENT_START',
      'GET_PROFILE_CONTENT_SUCCEEDED'
    ];
    const store = mockStore({});
    return store.dispatch(actions.getUserFollowers('e25af1ae-3835-46cb-a488-716936a07ab5'))
      .then(() => {
        const actualActions = store.getActions().map(action => action.type);
        expect(actualActions).toEqual(expectedActions);
      });
  });

  it('should create an action of type GET_PROFILE_CONTENT_START and GET_PROFILE_CONTENT_FAILED', () => {
    api.client.get = jest.fn().mockReturnValue(Promise.reject(new Error('An error occured')));
    const expectedActions = [
      'GET_PROFILE_CONTENT_START',
      'GET_PROFILE_CONTENT_FAILED',
    ];
    const store = mockStore({});
    return store.dispatch(actions.getUserFollowers('e25af1ae-3835-46cb-a488-716936a07ab5'))
      .then(() => {
        const actualActions = store.getActions().map(action => action.type);
        expect(actualActions).toEqual(expectedActions);
      });
  });

  it('should create an action of type GET_PROFILE_STARTED and GET_PROFILE_SUCCEEDED if successful', () => {
    api.client.get = jest.fn().mockReturnValue(Promise.resolve({
      data: {
        body: JSON.stringify({}),
        data: {
          articleCount: 6,
          articles: [{ }, { }, { }, {}, { }],
          currentPage: 1,
          pageCount: 1,
          status: 200
        }
      }
    }));

    const expectedActions = [
      'GET_PROFILE_STARTED',
      'GET_PROFILE_SUCCEEDED'
    ];
    const store = mockStore({});
    return store.dispatch(actions.getProfile('e25af1ae-3835-46cb-a488-716936a07ab5'))
      .then(() => {
        const actualActions = store.getActions().map(action => action.type);
        expect(actualActions).toEqual(expectedActions);
      });
  });

  it('should create an action of type GET_PROFILE_CONTENT_START & GET_PROFILE_CONTENT_FAILED', () => {
    api.client.get = jest.fn().mockReturnValue(Promise.reject(new Error('An error occured')));
    const expectedActions = [
      'GET_PROFILE_STARTED',
      'GET_PROFILE_FAILED',
    ];
    const store = mockStore({});
    return store.dispatch(actions.getProfile('2aec1bef'))
      .then(() => {
        const actualActions = store.getActions().map(action => action.type);
        expect(actualActions).toEqual(expectedActions);
      });
  });
});
