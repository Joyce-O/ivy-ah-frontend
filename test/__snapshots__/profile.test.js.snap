// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileArticleCard/> Should match the snapshot 1`] = `
<div
  className="column"
>
  <a
    href="/article/someid"
    onClick={[Function]}
  >
    <div
      className="profile-article"
      style={
        Object {
          "display": "block",
        }
      }
    >
      <div
        className="ui cards"
      >
        <div
          className="profile-article-wrapper"
        >
          <img
            className="ui avatar left floated image profile-article-user-image"
            src="someUrl"
          />
          <span
            className="profile-article-user-name"
          >
            undefined anothername
          </span>
          <p
            className="profile-article-readtime"
          >
             
            &lt; 1 min read
          </p>
        </div>
        <div
          style={
            Object {
              "backgroundImage": "url(undefined)",
              "backgroundPosition": "center",
              "backgroundRepeat": "no-repeat",
              "backgroundSize": "cover",
              "height": "10rem",
              "minWidth": "100%",
              "overflow": "hidden",
            }
          }
        >
          \`
        </div>
        <div
          className="w-100"
        >
          <div
            className="detail-container"
          >
            <p
              style={
                Object {
                  "color": "black",
                }
              }
            />
            <div
              className="article-hypes"
            >
              <span
                style={
                  Object {
                    "color": "blue",
                  }
                }
              >
                undefined hypes
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;

exports[`<ProfileFollowingCard/> Should match the snapshot 1`] = `
<div
  className="column"
>
  <a
    href="/profileRedirect/someid"
    onClick={[Function]}
  >
    <div
      style={
        Object {
          "marginBottom": "2rem",
        }
      }
    >
      <div
        className="ui fluid card"
        onClick={[Function]}
      >
        <div
          className="content"
        >
          <img
            className="ui small circular left floated image"
            src="someurl"
            style={
              Object {
                "marginRight": "2rem",
                "width": "6rem",
              }
            }
          />
          <div
            className="header"
            style={
              Object {
                "paddingBottom": "0.3rem",
                "paddingTop": "0.7rem",
              }
            }
          >
            somelastname anyname
          </div>
          <div
            className="meta"
            style={
              Object {
                "paddingBottom": "0.5rem",
              }
            }
          >
            joined Authors' haven 
          </div>
          <div
            className="meta"
          >
            Following
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;

exports[`<ProfileFollowingCard/> Should match the snapshot 2`] = `
<div
  className="profile"
>
  <img
    className="ui medium circular fluid image"
    src="someurl"
    style={
      Object {
        "margin": "0 auto",
      }
    }
  />
  <div
    className="ui huge header profile-username"
  >
    anyname somelastname
  </div>
  <p>
    Been on Authors' Haven
    <span
      style={
        Object {
          "display": "block",
        }
      }
    >
      since
       
      undefined NaN, NaN
    </span>
  </p>
  <button
    className="ui button btn blueButton"
    onClick={[Function]}
  >
    Follow
  </button>
</div>
`;
